src/main/java/io/cdap/plugin/gcp/dataplex/sink/DataplexBatchSink.java

```diff
@@ -405,18 +406,22 @@ protected void initOutput(BatchSinkContext context, BigQuery bigQuery, String ou
     Configuration configuration = new Configuration(baseConfiguration);
 
     // Build GCS storage path for this bucket output.
+    DatasetId datasetId = DatasetId.of(datasetProject, dataset);
     String temporaryGcsPath = BigQuerySinkUtils.getTemporaryGcsPath(bucket, runUUID.toString(), tableName);
-    BigQuerySinkUtils.configureOutput(configuration,
-      DatasetId.of(datasetProject, dataset),
-      tableName,
-      temporaryGcsPath,
-      fields);
+    BigQuerySinkUtils.configureOutput(configuration, datasetId, tableName, temporaryGcsPath, fields);
     // Both emitLineage and setOutputFormat internally try to create an external dataset if it does not already exist.
     // We call emitLineage before since it creates the dataset with schema which is used.
     List<String> fieldNames = fields.stream()
       .map(BigQueryTableFieldSchema::getName)
       .collect(Collectors.toList());
-    BigQuerySinkUtils.recordLineage(context, outputName, tableSchema, fieldNames);
+    String fqn = BigQueryUtil.getFQN(datasetProject, dataset, config.getTable());
+    String location = bigQuery.getDataset(datasetId).getLocation();
+    String referenceName = Strings.isNullOrEmpty(config.getReferenceName())
```

Comment:
For dataplex (Source & Sink), I don't think I can move this logic to the config class as the config class does not readily have access to dataset value for the BQ table. It's only used once in sink and once in source